diff --git a/modeling.py b/modeling.py
index fed5259..89ad361 100644
--- a/modeling.py
+++ b/modeling.py
@@ -27,6 +27,8 @@ import numpy as np
 import six
 import tensorflow as tf
 
+from inline import Here
+
 
 class BertConfig(object):
   """Configuration for `BertModel`."""
@@ -323,6 +325,7 @@ def get_assignment_map_from_checkpoint(tvars, init_checkpoint):
   for var in tvars:
     name = var.name
     m = re.match("^(.*):\\d+$", name)
+    Here().given(name, r'aaa:000').check_true(m is not None).check_eq(m.group(1), 'aaa')
     if m is not None:
       name = m.group(1)
     name_to_variable[name] = var
