diff --git a/src/main/java/com/alibaba/fastjson/util/IOUtils.java b/src/main/java/com/alibaba/fastjson/util/IOUtils.java
index 1269b2aef..591c8b792 100644
--- a/src/main/java/com/alibaba/fastjson/util/IOUtils.java
+++ b/src/main/java/com/alibaba/fastjson/util/IOUtils.java
@@ -32,6 +32,8 @@ import java.util.Properties;
 
 import com.alibaba.fastjson.JSONException;
 
+import org.inlinetest.Here;
+
 /**
  * @author wenshao[szujobs@hotmail.com]
  */
@@ -226,6 +228,7 @@ public class IOUtils {
             q = i / 100;
             // really: r = i - (q * 100);
             r = (int) (i - ((q << 6) + (q << 5) + (q << 2)));
+            new Here().given(i, 2147483648L).given(q, 21474836L).given(r, 0).checkEq(r, 48);
             i = q;
             buf[--charPos] = DigitOnes[r];
             buf[--charPos] = DigitTens[r];
@@ -247,6 +250,7 @@ public class IOUtils {
         // assert(i2 <= 65536, i2);
         for (;;) {
             q2 = (i2 * 52429) >>> (16 + 3);
+            new Here().given(i2, 30).given(q2, 0).checkEq(q2, 3);
             r = i2 - ((q2 << 3) + (q2 << 1)); // r = i2-(q2*10) ...
             buf[--charPos] = digits[r];
             i2 = q2;
