diff --git a/jadx-core/src/main/java/jadx/core/xmlgen/CommonBinaryParser.java b/jadx-core/src/main/java/jadx/core/xmlgen/CommonBinaryParser.java
index 3377f614..5d52aab9 100644
--- a/jadx-core/src/main/java/jadx/core/xmlgen/CommonBinaryParser.java
+++ b/jadx-core/src/main/java/jadx/core/xmlgen/CommonBinaryParser.java
@@ -3,6 +3,8 @@ package jadx.core.xmlgen;
 import java.io.IOException;
 import java.util.Arrays;
 
+import org.inlinetest.Here;
+
 public class CommonBinaryParser extends ParserConstants {
 	protected ParserStream is;
 
@@ -57,6 +59,7 @@ public class CommonBinaryParser extends ParserConstants {
 		}
 		if ((len & 0x80) != 0) {
 			len = (len & 0x7F) << 8 | strArray[start++] & 0xFF;
+                        new Here().given(len, 0x90).given(start, 0).given(strArray, "\u00e0\u004f\u00d0\u0020\u00ea\u003a\u0069\u0010\u00a2\u00d8\u0008\u0000\u002b\u0030\u0030\u009d".getBytes()).checkEq(len, 4291);
 		}
 		byte[] arr = Arrays.copyOfRange(strArray, start, start + len);
 		return new String(arr, ParserStream.STRING_CHARSET_UTF8);
