diff --git a/streams/src/main/java/org/apache/kafka/streams/state/internals/Murmur3.java b/streams/src/main/java/org/apache/kafka/streams/state/internals/Murmur3.java
index 5581a03648..ae169b3e61 100644
--- a/streams/src/main/java/org/apache/kafka/streams/state/internals/Murmur3.java
+++ b/streams/src/main/java/org/apache/kafka/streams/state/internals/Murmur3.java
@@ -16,6 +16,8 @@
  */
 package org.apache.kafka.streams.state.internals;
 
+import org.inlinetest.Here;
+
 /**
  * This class was taken from Hive org.apache.hive.common.util;
  * https://github.com/apache/hive/blob/master/storage-api/src/java/org/apache/hive/common/util/Murmur3.java
@@ -290,6 +292,7 @@ public class Murmur3 {
         switch (length - tailStart) {
             case 7:
                 k1 ^= ((long) data[offset + tailStart + 6] & 0xff) << 48;
+                new Here().given(k1, 0).given(data, "abcdefghijklmn".getBytes()).given(offset, 0).given(tailStart, 7).checkEq(k1, 30962247438172160L);
             case 6:
                 k1 ^= ((long) data[offset + tailStart + 5] & 0xff) << 40;
             case 5:
